app/models/resource_step.rb

```diff
@@ -0,0 +1,4 @@
+class ResourceStep < ApplicationRecord
+  belongs_to :resource
+  validates :number, uniqueness: { scope: :resource_id, message: "should be unique within each resource"}, presence: true
```

Comment:
One implication of this validation and the way the API is built right now is that it'll be kind of hard to update the step's number (e.g. if you have steps numbered 1, 2, 3, and you want to switch steps 1 and 3, you'd have to change step 1 to step 4, then step 3 to step 1, then step 1 to step 3). I'm assuming this is a pretty rare case, so I'm not optimizing for it, but let me know if that sounds like a problem. I could also set up a separate endpoint specifically for batch-changing step order.